api/v1beta2/awsmachine_types.go

```diff
@@ -185,6 +193,18 @@ type CloudInit struct {
 	SecureSecretsBackend SecretBackend `json:"secureSecretsBackend,omitempty"`
 }
 
+// AWSInstanceDetails describes an instance details for creating a AWSMachine.
+type AWSInstanceDetails struct {
+	// InstanceType is the type of instance to create. Example: m4.xlarge
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength:=2
+	InstanceType string `json:"instanceType"`
+
+	// SpotMarketOptions allows users to configure instances to be run using AWS Spot instances.
+	// +optional
+	SpotMarketOptions *SpotMarketOptions `json:"spotMarketOptions,omitempty"`
```

Comment:
Do you think `SpotMarketOptions` here is redundant with the one already defined in the machine spec? Is the idea that each `InstanceType` would have its own `SpotMarketOptions`. Some clarification here would be nice.